runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -269,72 +270,145 @@ public void onMenuEntryAdded(MenuEntryAdded event)
-					.map(ColorTileMarker::getColor)
-					.distinct()
-					.collect(Collectors.toList());
-				for (Color color : existingColors)
+					.setParent(borderColorMenu)
+					.onClick(e -> setTileBorderColor(existingPoint, color));
+			}
+		}
+
+		client.createMenuEntry(idx--)
+			.setOption("Pick color")
+			.setType(MenuAction.RUNELITE)
+			.setParent(borderColorMenu)
+			.onClick(e ->
+			{
+				Color color = existingPoint.getBorderColor();
+				SwingUtilities.invokeLater(() ->
 				{
-					if (!color.equals(existing.getColor()))
-					{
-						client.createMenuEntry(-3)
-							.setOption(ColorUtil.prependColorTag("Color", color))
-							.setType(MenuAction.RUNELITE)
-							.setParent(menuColor)
-							.onClick(e -> colorTile(existing, color));
-					}
-				}
+					RuneliteColorPicker colorPicker = colorPickerManager.create(SwingUtilities.windowForComponent((Applet) client),
+						color, "Tile border color", false);
+					colorPicker.setOnClose(c -> setTileBorderColor(existingPoint, c));
+					colorPicker.setVisible(true);
+				});
+			});
+
+		if (regionPoints.size() > 1)
+		{
+			client.createMenuEntry(idx--)
+				.setOption("Reset all border colors")
+				.setType(MenuAction.RUNELITE)
+				.setParent(borderColorMenu)
+				.onClick(e ->
+					chatboxPanelManager.openTextMenuInput("Are you sure you want to reset the border color of " + regionPoints.size() + " tiles?")
+						.option("Yes", () ->
+						{
+							var newPoints = regionPoints.stream()
+								.map(p -> new GroundMarkerPoint(p.getRegionId(), p.getRegionX(), p.getRegionY(), p.getZ(), config.borderColor(), p.getFillColor(), p.getLabel()))
+								.collect(Collectors.toList());
+							savePoints(regionId, newPoints);
+							loadPoints();
+						})
+						.option("No", Runnables.doNothing())
+						.build());
+		}
+
+		return idx;
+	}
+
+	private int createTileFillColorMenu(int idx, GroundMarkerPoint existingPoint, int regionId, Collection<GroundMarkerPoint> regionPoints){
+		MenuEntry fillColorMenu = client.createMenuEntry(idx--)
+			.setOption("Mark fill color")
+			.setTarget("Tile")
+			.setType(MenuAction.RUNELITE_SUBMENU);
+
+		var existingColors = points.stream()
+			.map(ColorTileMarker::getFillColor)
+			.distinct()
+			.collect(Collectors.toList());
+
+		for (Color color : existingColors)
+		{
+			if (color != null && !color.equals(existingPoint.getFillColor()))
+			{
+				client.createMenuEntry(idx--)
+					.setOption(ColorUtil.prependColorTag("Set Color", color))
+					.setType(MenuAction.RUNELITE)
+					.setParent(fillColorMenu)
+					.onClick(e -> setTileFillColor(existingPoint, color));
 			}
 		}
+
+		client.createMenuEntry(idx--)
+			.setOption("Pick color")
+			.setType(MenuAction.RUNELITE)
+			.setParent(fillColorMenu)
+			.onClick(e ->
+			{
+				Color color = existingPoint.getFillColor();
+				SwingUtilities.invokeLater(() ->
+				{
+					RuneliteColorPicker colorPicker = colorPickerManager.create(SwingUtilities.windowForComponent((Applet) client),
```

Comment:
Selecting this menu entry for tiles created before this version of the plugin results in
``2024-03-10 12:22:17 EDT [AWT-EventQueue-0] ERROR net.runelite.client.RuneLite - Uncaught exception:
java.lang.NullPointerException: Cannot invoke "java.awt.Color.getRed()" because "target" is null
	at net.runelite.client.ui.components.colorpicker.HuePanel.closestYToColor(HuePanel.java:116)
	at net.runelite.client.ui.components.colorpicker.HuePanel.select(HuePanel.java:88)
	at net.runelite.client.ui.components.colorpicker.RuneliteColorPicker.updatePanels(RuneliteColorPicker.java:331)
	at net.runelite.client.ui.components.colorpicker.RuneliteColorPicker.<init>(RuneliteColorPicker.java:299)
	at net.runelite.client.ui.components.colorpicker.ColorPickerManager.create(ColorPickerManager.java:56)
	at net.runelite.client.plugins.groundmarkers.GroundMarkerPlugin.lambda$createTileFillColorMenu$15(GroundMarkerPlugin.java:387)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)``

Due to color being null, should check if null and if so default to the config value

Reply:
It would probably be in our best interest to update the color picker to not crash and burn on null color values, or call out that it needs a nonnull input.